tests/integration/api_v3/rpc/generators.test.js

```diff
@@ -79,48 +82,113 @@ describe('Generators API', () => {
 		});
 
-		it('returns generators list with limit=100 -> ok', async () => {
+		it('should return generators list when called with limit=100', async () => {
 			const response = await getGenerators({ limit: 100 });
 			expect(response).toMap(jsonRpcEnvelopeSchema);
 			const { result } = response;
-			expect(result).toMap(resultEnvelopeSchema);
+			expect(result).toMap(generatorResponseSchema);
 			expect(result.data).toBeInstanceOf(Array);
 			expect(result.data.length).toBeGreaterThanOrEqual(1);
 			expect(result.data.length).toBeLessThanOrEqual(100);
-			result.data.map(generator => expect(generator).toMap(generatorSchema));
 			expect(result.meta).toMap(metaSchema);
 		});
 
-		it('returns generators list with limit=100 and offset=1 -> ok', async () => {
+		it('should return generators list when called with limit=100 and offset=1', async () => {
 			const response = await getGenerators({ limit: 100, offset: 1 });
 			expect(response).toMap(jsonRpcEnvelopeSchema);
 			const { result } = response;
-			expect(result).toMap(resultEnvelopeSchema);
+			expect(result).toMap(generatorResponseSchema);
 			expect(result.data).toBeInstanceOf(Array);
 			expect(result.data.length).toBeGreaterThanOrEqual(1);
 			expect(result.data.length).toBeLessThanOrEqual(100);
-			result.data.map(generator => expect(generator).toMap(generatorSchema));
 			expect(result.meta).toMap(metaSchema);
 		});
 
-		it('empty limit -> all generators', async () => {
+		it('should return generators list when called with empty limit', async () => {
 			const response = await getGenerators({ limit: '' });
 			expect(response).toMap(jsonRpcEnvelopeSchema);
 			const { result } = response;
-			expect(result).toMap(resultEnvelopeSchema);
+			expect(result).toMap(generatorResponseSchema);
 			expect(result.data).toBeInstanceOf(Array);
 			expect(result.data.length).toBeGreaterThanOrEqual(1);
 			expect(result.data.length).toBeLessThanOrEqual(103);
-			result.data.map(generator => expect(generator).toMap(generatorSchema));
 			expect(result.meta).toMap(metaSchema);
 		});
 
-		it('limit=0 -> -32602', async () => {
+		it('should return generators list when searching with generator name', async () => {
+			const response = await getGenerators({ search: firstGenerator.name });
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result).toMap(generatorResponseSchema);
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBe(1);
+			expect(result.meta).toMap(metaSchema);
+		});
+
+		it('should return generators list when searching with generator address', async () => {
+			const response = await getGenerators({ search: firstGenerator.address });
+			expect(response).toMap(jsonRpcEnvelopeSchema);
+			const { result } = response;
+			expect(result).toMap(generatorResponseSchema);
+			expect(result.data).toBeInstanceOf(Array);
+			expect(result.data.length).toBe(1);
+			expect(result.meta).toMap(metaSchema);
```

Comment:
Following checks are already covered by schema validation. Same for other tests.

```suggestion
			expect(result.data.length).toBe(1);
```